homeassistant/components/moehlenhoff_alpha2/__init__.py

```diff
@@ -0,0 +1,108 @@
+"""Support for the Moehlenhoff Alpha2."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import aiohttp
+from moehlenhoff_alpha2 import Alpha2Base
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import DOMAIN, SIGNAL_HEATAREA_DATA_UPDATED
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+UPDATE_INTERVAL = timedelta(seconds=60)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up a config entry."""
+    base = Alpha2Base(entry.data["host"])
+    try:
+        await base.update_data()
+    except (
+        aiohttp.client_exceptions.ClientConnectorError,
+        asyncio.TimeoutError,
+    ) as err:
+        raise exceptions.ConfigEntryNotReady from err
+
+    base_uh = Alpha2BaseUpdateHandler(hass, base)
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {"connection": base_uh, "devices": set()}
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    # Trigger updates at regular intervals.
+    async_track_time_interval(hass, base_uh.async_update, UPDATE_INTERVAL)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    if unload_ok and entry.entry_id in hass.data[DOMAIN]:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry):
```

Comment:
Please type the whole signature when adding type annotations.
```suggestion
async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
```